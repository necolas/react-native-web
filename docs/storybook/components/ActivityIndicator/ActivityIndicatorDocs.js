/* eslint-disable react/jsx-sort-props */

/**
 * @flow
 */

import PropAnimating from './examples/PropAnimating';
import PropColor from './examples/PropColor';
import PropHidesWhenStopped from './examples/PropHidesWhenStopped';
import PropSize from './examples/PropSize';
import React from 'react';
import { storiesOf } from '@kadira/storybook';
import UIExplorer, { DocItem } from '../../ui-explorer';

const sections = [
  {
    title: 'Props',
    entries: [
      <DocItem name="...View props" />,

      <DocItem
        name="animating"
        typeInfo="?boolean = true"
        description="Whether to show the indicator or hide it."
        example={{
          render: () => <PropAnimating />
        }}
      />,

      <DocItem
        name="color"
        typeInfo="?color = #1976D2"
        description="The foreground color of the spinner."
        example={{
          render: () => <PropColor />
        }}
      />,

      <DocItem
        name="hidesWhenStopped"
        typeInfo="?boolean = true"
        description="Whether the indicator should hide when not animating."
        example={{
          render: () => <PropHidesWhenStopped />
        }}
      />,

      <DocItem
        name="size"
        typeInfo="?enum('small', 'large') | number = 'small'"
        description="Size of the indicator. Small has a height of 20px, large has a height of 36px."
        example={{
          render: () => <PropSize />
        }}
      />
    ]
  }
];

storiesOf('Components', module).add('ActivityIndicator', () =>
  <UIExplorer
    description="Displays a customizable activity indicator"
    sections={sections}
    title="ActivityIndicator"
    url="components/ActivityIndicator"
  />
);
