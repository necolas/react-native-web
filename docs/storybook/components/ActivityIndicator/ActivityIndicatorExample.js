/* eslint-disable react/prefer-es6-class, react/prop-types */

/**
 * Copyright (c) 2013-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * The examples provided by Facebook are for non-commercial testing and
 * evaluation purposes only.
 *
 * Facebook reserves all rights not expressly granted.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL
 * FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
 * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 *
 * @flow
 */

import createReactClass from 'create-react-class';
import React from 'react';
import { storiesOf } from '@kadira/storybook';
import TimerMixin from 'react-timer-mixin';
import UIExplorer from '../../UIExplorer';
import { ActivityIndicator, StyleSheet, View } from 'react-native';

const ToggleAnimatingActivityIndicator = createReactClass({
  mixins: [TimerMixin],

  getInitialState() {
    return {
      animating: true
    };
  },

  setToggleTimeout() {
    this.setTimeout(() => {
      this.setState({ animating: !this.state.animating });
      this.setToggleTimeout();
    }, 2000);
  },

  componentDidMount() {
    this.setToggleTimeout();
  },

  render() {
    return (
      <ActivityIndicator
        animating={this.state.animating}
        hidesWhenStopped={this.props.hidesWhenStopped}
        size="large"
        style={styles.centering}
      />
    );
  }
});

const examples = [
  {
    title: 'Default',
    description: 'Renders small and blue',
    render() {
      return (
        <View style={styles.horizontal}>
          <ActivityIndicator />
        </View>
      );
    }
  },
  {
    title: 'Custom colors',
    description: 'Any color value is supported',
    render() {
      return (
        <View style={styles.horizontal}>
          <ActivityIndicator color="#1DA1F2" style={styles.rightPadding} />
          <ActivityIndicator color="#17BF63" style={styles.rightPadding} />
          <ActivityIndicator color="#F45D22" style={styles.rightPadding} />
          <ActivityIndicator color="#794BC4" style={styles.rightPadding} />
          <ActivityIndicator color="#E0245E" style={styles.rightPadding} />
          <ActivityIndicator color="#FFAD1F" style={styles.rightPadding} />
        </View>
      );
    }
  },
  {
    title: 'Custom sizes',
    description:
      'There are 2 predefined sizes: "small" and "large". The size can be further customized as a number (pixels) or using scale transforms',
    render() {
      return (
        <View style={styles.horizontal}>
          <ActivityIndicator size={20} style={styles.rightPadding} />
          <ActivityIndicator size="small" style={styles.rightPadding} />
          <ActivityIndicator size={36} style={styles.rightPadding} />
          <ActivityIndicator size="large" style={styles.rightPadding} />
          <ActivityIndicator size={60} style={styles.rightPadding} />
          <ActivityIndicator
            size="large"
            style={{ marginLeft: 20, transform: [{ scale: 1.75 }] }}
          />
        </View>
      );
    }
  },
  {
    title: 'Animation controls (start, pause, hide)',
    description: 'The animation can be paused, and the component hidden',
    render() {
      return (
        <View style={[styles.horizontal]}>
          <ToggleAnimatingActivityIndicator hidesWhenStopped={false} style={styles.rightPadding} />
          <ToggleAnimatingActivityIndicator />
        </View>
      );
    }
  }
];

const styles = StyleSheet.create({
  gray: {
    backgroundColor: '#cccccc'
  },
  horizontal: {
    alignItems: 'center',
    flexDirection: 'row'
  },
  rightPadding: {
    paddingRight: 10
  }
});

storiesOf('Components', module).add('ActivityIndicator', () =>
  <UIExplorer
    description="Displays a customizable activity indicator"
    examples={examples}
    title="ActivityIndicator"
    url="https://github.com/necolas/react-native-web/blob/master/docs/components/ActivityIndicator.md"
  />
);
