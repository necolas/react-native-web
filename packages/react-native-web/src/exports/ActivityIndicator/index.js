/**
 * Copyright (c) Nicolas Gallagher.
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

import type { ViewProps } from '../View';

import StyleSheet from '../StyleSheet';
import View from '../View';
import React, { forwardRef } from 'react';

const accessibilityValue = { max: 1, min: 0 };

const createSvgCircle = (style) => (
  <circle cx="16" cy="16" fill="none" r="14" strokeWidth="4" style={style} />
);

type ActivityIndicatorProps = {
  ...ViewProps,
  animating?: boolean,
  color?: ?string,
  hidesWhenStopped?: boolean,
  size?: 'small' | 'large' | number
};

const ActivityIndicator = forwardRef<ActivityIndicatorProps, *>((props, forwardedRef) => {
  const {
    animating = true,
    color = '#1976D2',
    hidesWhenStopped = true,
    size = 'small',
    style,
    ...other
  } = props;

  const svg = (
    <svg height="100%" viewBox="0 0 32 32" width="100%">
      {createSvgCircle({
        stroke: color,
        opacity: 0.2
      })}
      {createSvgCircle({
        stroke: color,
        strokeDasharray: 80,
        strokeDashoffset: 60
      })}
    </svg>
  );

  return (
    <View
      {...other}
      accessibilityRole="progressbar"
      accessibilityValue={accessibilityValue}
      ref={forwardedRef}
      style={[styles.container, style]}
    >
      <View
        children={svg}
        style={[
          typeof size === 'number' ? { height: size, width: size } : indicatorSizes[size],
          styles.animation,
          !animating && styles.animationPause,
          !animating && hidesWhenStopped && styles.hidesWhenStopped
        ]}
      />
    </View>
  );
});

ActivityIndicator.displayName = 'ActivityIndicator';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center'
  },
  hidesWhenStopped: {
    visibility: 'hidden'
  },
  animation: {
    animationDuration: '0.75s',
    animationKeyframes: [
      {
        '0%': { transform: [{ rotate: '0deg' }] },
        '100%': { transform: [{ rotate: '360deg' }] }
      }
    ],
    animationTimingFunction: 'linear',
    animationIterationCount: 'infinite'
  },
  animationPause: {
    animationPlayState: 'paused'
  }
});

const indicatorSizes = StyleSheet.create({
  small: {
    width: 20,
    height: 20
  },
  large: {
    width: 36,
    height: 36
  }
});

export default ActivityIndicator;
