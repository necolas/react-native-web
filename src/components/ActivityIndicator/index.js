import applyNativeMethods from '../../modules/applyNativeMethods';
import StyleSheet from '../../apis/StyleSheet';
import View from '../View';
import React, { Component, PropTypes } from 'react';

class ActivityIndicator extends Component {
  static displayName = 'ActivityIndicator';

  static propTypes = {
    ...View.propTypes,
    animating: PropTypes.bool,
    color: PropTypes.string,
    hidesWhenStopped: PropTypes.bool,
    size: PropTypes.oneOfType([ PropTypes.oneOf([ 'small', 'large' ]), PropTypes.number ])
  };

  static defaultProps = {
    animating: true,
    color: '#1976D2',
    hidesWhenStopped: true,
    size: 'small'
  };

  render() {
    const {
      animating,
      color,
      hidesWhenStopped,
      size,
      style,
      ...other
    } = this.props;

    const svg = (
      <svg height='100%' viewBox='0 0 32 32' width='100%'>
        <circle
          cx='16'
          cy='16'
          fill='none'
          r='14'
          strokeWidth='4'
          style={{
            stroke: color,
            opacity: 0.2
          }}
        />
        <circle
          cx='16'
          cy='16'
          fill='none'
          r='14'
          strokeWidth='4'
          style={{
            stroke: color,
            strokeDasharray: 80,
            strokeDashoffset: 60
          }}
        />
      </svg>
    );

    return (
      <View {...other}
        accessibilityRole='progressbar'
        aria-valuemax='1'
        aria-valuemin='0'
        style={[
          styles.container,
          style,
          typeof size === 'number' && { height: size, width: size }
        ]}
      >
        <View
          children={svg}
          style={[
            indicatorSizes[size],
            styles.animation,
            !animating && styles.animationPause,
            !animating && hidesWhenStopped && styles.hidesWhenStopped
          ]}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center'
  },
  hidesWhenStopped: {
    visibility: 'hidden'
  },
  animation: {
    animationDuration: '0.75s',
    animationName: 'rn-ActivityIndicator-animation',
    animationTimingFunction: 'linear',
    animationIterationCount: 'infinite'
  },
  animationPause: {
    animationPlayState: 'paused'
  }
});

const indicatorSizes = StyleSheet.create({
  small: {
    width: 20,
    height: 20
  },
  large: {
    width: 36,
    height: 36
  }
});

module.exports = applyNativeMethods(ActivityIndicator);
