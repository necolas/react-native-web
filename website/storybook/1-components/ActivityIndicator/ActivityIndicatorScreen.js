/* eslint-disable react/jsx-sort-props */

/**
 * @flow
 */

import PropAnimating from './examples/PropAnimating';
import PropColor from './examples/PropColor';
import PropHidesWhenStopped from './examples/PropHidesWhenStopped';
import PropSize from './examples/PropSize';
import React from 'react';
import UIExplorer, { Description, DocItem, Section, storiesOf } from '../../ui-explorer';

const ActivityIndicatorScreen = () => (
  <UIExplorer title="ActivityIndicator" url="1-components/ActivityIndicator">
    <Description>Displays a customizable activity indicator</Description>
    <Section title="Props">
      <DocItem name="...View props" />

      <DocItem
        name="animating"
        typeInfo="?boolean = true"
        description="Whether to show the indicator or hide it."
        example={{
          render: () => <PropAnimating />
        }}
      />

      <DocItem
        name="color"
        typeInfo="?color = #1976D2"
        description="The foreground color of the spinner."
        example={{
          render: () => <PropColor />
        }}
      />

      <DocItem
        name="hidesWhenStopped"
        typeInfo="?boolean = true"
        description="Whether the indicator should hide when not animating."
        example={{
          render: () => <PropHidesWhenStopped />
        }}
      />

      <DocItem
        name="size"
        typeInfo="?enum('small', 'large') | number = 'small'"
        description="Size of the indicator. Small has a height of 20px, large has a height of 36px."
        example={{
          render: () => <PropSize />
        }}
      />
    </Section>
  </UIExplorer>
);

storiesOf('Components', module).add('ActivityIndicator', ActivityIndicatorScreen);
