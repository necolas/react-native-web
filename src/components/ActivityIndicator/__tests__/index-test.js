/* eslint-env jasmine, jest */

import ActivityIndicator from '..';
import React from 'react';
import { render } from 'enzyme';

describe('components/ActivityIndicator', () => {
  describe('prop "animating"', () => {
    test('is "true"', () => {
      const component = render(<ActivityIndicator animating={true} />);
      expect(component).toMatchSnapshot();
    });

    test('is "false"', () => {
      const component = render(<ActivityIndicator animating={false} />);
      expect(component).toMatchSnapshot();
    });
  });

  test('prop "color"', () => {
    const component = render(<ActivityIndicator color="red" />);
    expect(component).toMatchSnapshot();
  });

  describe('prop "hidesWhenStopped"', () => {
    test('is "true"', () => {
      const component = render(<ActivityIndicator animating={false} hidesWhenStopped={true} />);
      expect(component).toMatchSnapshot();
    });

    test('is "false"', () => {
      const component = render(<ActivityIndicator animating={false} hidesWhenStopped={false} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('prop "size"', () => {
    test('is "large"', () => {
      const component = render(<ActivityIndicator size="large" />);
      expect(component).toMatchSnapshot();
    });

    test('is a number', () => {
      const component = render(<ActivityIndicator size={30} />);
      expect(component).toMatchSnapshot();
    });
  });
});
