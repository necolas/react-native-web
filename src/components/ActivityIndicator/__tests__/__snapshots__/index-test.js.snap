// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ActivityIndicator prop "animating" is "false" 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
  role="progressbar"
>
  <div
    class="rn-alignItems-1oszu61 rn-animationDuration-17bb2tj rn-animationIterationCount-1muvv40 rn-animationName-dozj4v rn-animationPlayState-1abnn5w rn-animationTimingFunction-1ldzwu0 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-z80fyv rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-visibility-11j9u27 rn-width-19wmn03"
  >
    <svg
      height="100%"
      viewbox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:#1976D2;opacity:0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:#1976D2;stroke-dasharray:80;stroke-dashoffset:60;"
      />
    </svg>
  </div>
</div>
`;

exports[`components/ActivityIndicator prop "animating" is "true" 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
  role="progressbar"
>
  <div
    class="rn-alignItems-1oszu61 rn-animationDuration-17bb2tj rn-animationIterationCount-1muvv40 rn-animationName-dozj4v rn-animationTimingFunction-1ldzwu0 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-z80fyv rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-width-19wmn03"
  >
    <svg
      height="100%"
      viewbox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:#1976D2;opacity:0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:#1976D2;stroke-dasharray:80;stroke-dashoffset:60;"
      />
    </svg>
  </div>
</div>
`;

exports[`components/ActivityIndicator prop "color" 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
  role="progressbar"
>
  <div
    class="rn-alignItems-1oszu61 rn-animationDuration-17bb2tj rn-animationIterationCount-1muvv40 rn-animationName-dozj4v rn-animationTimingFunction-1ldzwu0 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-z80fyv rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-width-19wmn03"
  >
    <svg
      height="100%"
      viewbox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:red;opacity:0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:red;stroke-dasharray:80;stroke-dashoffset:60;"
      />
    </svg>
  </div>
</div>
`;

exports[`components/ActivityIndicator prop "hidesWhenStopped" is "false" 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
  role="progressbar"
>
  <div
    class="rn-alignItems-1oszu61 rn-animationDuration-17bb2tj rn-animationIterationCount-1muvv40 rn-animationName-dozj4v rn-animationPlayState-1abnn5w rn-animationTimingFunction-1ldzwu0 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-z80fyv rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-width-19wmn03"
  >
    <svg
      height="100%"
      viewbox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:#1976D2;opacity:0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:#1976D2;stroke-dasharray:80;stroke-dashoffset:60;"
      />
    </svg>
  </div>
</div>
`;

exports[`components/ActivityIndicator prop "hidesWhenStopped" is "true" 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
  role="progressbar"
>
  <div
    class="rn-alignItems-1oszu61 rn-animationDuration-17bb2tj rn-animationIterationCount-1muvv40 rn-animationName-dozj4v rn-animationPlayState-1abnn5w rn-animationTimingFunction-1ldzwu0 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-z80fyv rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-visibility-11j9u27 rn-width-19wmn03"
  >
    <svg
      height="100%"
      viewbox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:#1976D2;opacity:0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:#1976D2;stroke-dasharray:80;stroke-dashoffset:60;"
      />
    </svg>
  </div>
</div>
`;

exports[`components/ActivityIndicator prop "size" is "large" 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
  role="progressbar"
>
  <div
    class="rn-alignItems-1oszu61 rn-animationDuration-17bb2tj rn-animationIterationCount-1muvv40 rn-animationName-dozj4v rn-animationTimingFunction-1ldzwu0 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-1r8g8re rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-width-1acpoxo"
  >
    <svg
      height="100%"
      viewbox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:#1976D2;opacity:0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:#1976D2;stroke-dasharray:80;stroke-dashoffset:60;"
      />
    </svg>
  </div>
</div>
`;

exports[`components/ActivityIndicator prop "size" is a number 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
  role="progressbar"
>
  <div
    class="rn-alignItems-1oszu61 rn-animationDuration-17bb2tj rn-animationIterationCount-1muvv40 rn-animationName-dozj4v rn-animationTimingFunction-1ldzwu0 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
    style="height:30px;width:30px;"
  >
    <svg
      height="100%"
      viewbox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:#1976D2;opacity:0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke:#1976D2;stroke-dasharray:80;stroke-dashoffset:60;"
      />
    </svg>
  </div>
</div>
`;
